File: apps/lesson-plan-generator/internal/pipeline/narration.go
package pipeline

import "lesson-plan-generator/internal/domain"

// step 2 – split narration into segments.
// Currently returns the pre-defined segments from the lesson.
// Future: accept raw narration text and auto-split on sentence boundaries.
func splitNarration(lesson domain.LessonContent) []domain.NarrationSegment {
	return lesson.Narration
}

File: apps/lesson-plan-generator/app/steps/narration.py
"""Step 2 – Split narration into segments.

Currently returns the pre-defined segments from the lesson fixture.
Future: accept a single narration string and auto-split on sentence boundaries.
"""

from app.models import LessonContent, NarrationSegment


def split_narration(lesson: LessonContent) -> list[NarrationSegment]:
    return lesson.narration
